sceneview/src/main/kotlin/io/github/sceneview/interaction/CameraGestureHandler.kt

```diff
@@ -0,0 +1,64 @@
+package io.github.sceneview.interaction
+
+import android.view.MotionEvent
+import com.google.android.filament.utils.Manipulator
+import dev.romainguy.kotlin.math.lookAt
+import io.github.sceneview.SceneLifecycleObserver
+import io.github.sceneview.SceneView
+import io.github.sceneview.math.toFloat3
+import io.github.sceneview.node.Node
+import io.github.sceneview.utils.FrameTime
+
+class CameraGestureHandler(private val sceneView: SceneView) : GestureHandler(sceneView),
+    SceneLifecycleObserver {
+    private val manipulator = Manipulator.Builder().build(Manipulator.Mode.ORBIT)
+    private val gestureDetector = GestureDetector(view, FilamentManipulatorAdapter(manipulator))
+
+    private val eyePos = FloatArray(3)
+    private val target = FloatArray(3)
+    private val upward = FloatArray(3)
+
+    init {
+        sceneView.lifecycle.addObserver(this)
+    }
+
+    override fun onTouchEvent(event: MotionEvent) {
+        gestureDetector.onTouchEvent(event)
+    }
+
+    override fun onFrame(frameTime: FrameTime) {
+        super.onFrame(frameTime)
+        manipulator.getLookAt(eyePos, target, upward)
+        sceneView.camera.transform = lookAt(
+            eyePos.toFloat3(),
+            target.toFloat3(),
+            upward.toFloat3()
+        )
```

Comment:
@ThomasGorisse Could you help me with this part? The original part in Filament's [ModelViewer ](https://github.com/google/filament/blob/f56f5a300c81321a578dce1b2b6f1fd37788c0ab/android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt#L286) is 
```
 cameraManipulator.getLookAt(eyePos, target, upward)
        camera.lookAt(
                eyePos[0], eyePos[1], eyePos[2],
                target[0], target[1], target[2],
                upward[0], upward[1], upward[2])
```
Accessing the Filament Camera didn't work for me because the position is updated afterwards with the SceneView Camera.

Reply:
There were an issue within the `lookAt()` function of kotlin-math.
Can you copy paste the class form here to SceneView math package?
https://github.com/romainguy/kotlin-math/tree/main/src/commonMain/kotlin/dev/romainguy/kotlin/math

Reply:
@ThomasGorisse With the updated kotlin-math files everything looks much better. I used the model-viewer sample app to verify it and although the gestures work I'm wondering whether it is 100% correct: In my understanding the camera should move around the targetPosition (origin by default). But it looks like it's moving around its own position. Could you take a look again? Thanks :)